tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/WebRequestTests.cs

```diff
@@ -84,34 +86,34 @@ private async Task RunTest(string metadataSchemaVersion)
-            {
-                var allSpans = agent.WaitForSpans(expectedAllSpansCount).OrderBy(s => s.Start).ToList();
-                allSpans.Should().OnlyHaveUniqueItems(s => new { s.SpanId, s.TraceId });
-
-                var spans = allSpans.Where(s => s.Type == SpanTypes.Http).ToList();
-                spans.Should().HaveCount(expectedSpanCount);
-                ValidateIntegrationSpans(spans, metadataSchemaVersion, expectedServiceName: clientSpanServiceName, isExternalSpan);
-
-                var okSpans = spans.Where(s => s.Tags[Tags.HttpStatusCode] == "200").ToList();
-                var notFoundSpans = spans.Where(s => s.Tags[Tags.HttpStatusCode] == "404").ToList();
-                var teapotSpans = spans.Where(s => s.Tags[Tags.HttpStatusCode] == "418").ToList();
-
-                (okSpans.Count + notFoundSpans.Count + teapotSpans.Count).Should().Be(expectedSpanCount);
-                okSpans.Should().OnlyContain(s => s.Error == 0);
-                notFoundSpans.Should().OnlyContain(s => s.Error == 0);
-                teapotSpans.Should().OnlyContain(s => s.Error == 1);
-
-                var firstSpan = spans.First();
-                var traceId = StringUtil.GetHeader(processResult.StandardOutput, HttpHeaderNames.TraceId);
-                var parentSpanId = StringUtil.GetHeader(processResult.StandardOutput, HttpHeaderNames.ParentId);
-
-                Assert.Equal(firstSpan.TraceId.ToString(CultureInfo.InvariantCulture), traceId);
-                Assert.Equal(firstSpan.SpanId.ToString(CultureInfo.InvariantCulture), parentSpanId);
-                telemetry.AssertIntegrationEnabled(IntegrationId.WebRequest);
-                VerifyInstrumentation(processResult.Process);
-            }
+            using var agent = EnvironmentHelper.GetMockAgent();
+            using ProcessResult processResult = await RunSampleAndWaitForExit(agent, arguments: $"Port={httpPort}");
+
+            var allSpans = agent.WaitForSpans(expectedAllSpansCount).OrderBy(s => s.Start).ToList();
+
+            var settings = VerifyHelper.GetSpanVerifierSettings();
+#if NETCOREAPP
+            // different TFMs use different underlying handlers, which we don't really care about for the snapshots
+            settings.AddSimpleScrubber("System.Net.Http.HttpClientHandler", "System.Net.Http.SocketsHttpHandler");
+#endif
+            var suffix = EnvironmentHelper.IsCoreClr() ? string.Empty : "_netfx";
+            await VerifyHelper.VerifySpans(
+                                   allSpans,
+                                   settings,
+                                   spans =>
+                                       spans.OrderBy(x => VerifyHelper.GetRootSpanResourceName(x, spans))
+                                            .ThenBy(x => VerifyHelper.GetSpanDepth(x, spans))
+                                            .ThenBy(x => x.Tags.TryGetValue("http.url", out var url) ? url : string.Empty)
```

Comment:
A bit annoying, but we need to add this extra sorting layer to get stable snapshots